src/map/script.cpp

```diff
@@ -12296,12 +12299,27 @@ BUILDIN_FUNC(setmapflag)
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	if( mf == MF_SKILL_DAMAGE ){
-		ShowError( "buildin_setmapflag: Skill damage adjust is not supported. Please use setmapflagskilldmg.\n" );
-		return SCRIPT_CMD_FAILURE;
-	}
+	union u_mapflag_args args = {};
 
-	FETCH(4, args.flag_val);
+	switch(mf) {
+		case MF_SKILL_DAMAGE:
+			if (script_hasdata(st, 4) && script_hasdata(st, 5))
+				args.rate[script_getnum(st, 5)] = script_getnum(st, 4);
```

Comment:
This line causes an error in compile.

![image](https://user-images.githubusercontent.com/5693492/43018708-07d9ab24-8c8d-11e8-9521-e47255983c99.png)